service/iot-service-samples/azure-sas-credential-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/AzureSasCredentialSample.java

```diff
@@ -0,0 +1,354 @@
+        // https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-security#security-token-structure
+        AzureSasCredential credential = new AzureSasCredential(sharedAccessSignature);
+
+        // "my-azure-iot-hub.azure-devices.net" for example
+        String iotHubHostName = parsedArguments.getIotHubHostName();
+
+        String newDeviceId = runRegistryManagerSample(iotHubHostName, credential);
+
+        runTwinClientSample(iotHubHostName, credential, newDeviceId);
+
+        runServiceClientSample(iotHubHostName, credential, newDeviceId);
+
+        // The AzureSasCredential object supports updating the shared access signature which is important to do
+        // once the previous signature has expired, or is close to expiring. This sample assumes that no renewal will
+        // be necessary since the sample completes in a few seconds. After updating the credential object, any
+        // clients that were constructed with that credential will start using the updated signature
+
+        //String updatedSignature = "some new shared access signature";
+        //credential.update(updatedSignature);
+
+        runJobClientSample(iotHubHostName, credential);
+
+        runDeviceMethodClientSample(iotHubHostName, credential, newDeviceId);
+    }
+
+    /**
+     * Create a shared access signature for a given resource and key. This method isn't used by the sample,
+     * but it is a good reference for how to create shared access signatures in Java.
+     *
+     * @param resourceUri The resource that the shared access token should grant access to. For cases where the token
+     * will be used for more than one function (i.e. used by registryManager to create a device and used by serviceClient
+     * to send cloud to device messages), this value should be the hostName of your IoT Hub
+     * ("my-azure-iot-hub.azure-devices.net" for example). Shared access signatures do support scoping of the resource
+     * authorization by making this resourceUri more specific. For example, a resourceUri of "my-azure-iot-hub.azure-devices.net/devices"
+     * will make this token only usable when creating/updating/deleting device identities.
+     * For more examples, see <a href="https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-security#use-security-tokens-from-service-components">Using security tokens from service components</a>
+     * @param key The shared access key for your IoT Hub.
+     * @return The shared access signature. This value can be used as is to build a {@link AzureSasCredential} instance
+     * @throws UnsupportedEncodingException If UTF_8 is not a supported encoding on your device.
+     * @throws NoSuchAlgorithmException If HmacSHA256 algorithm isn't found.
+     * @throws InvalidKeyException If initializing the HmacSHA256 SHA fails.
+     * @see <a href="docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-security">Control access to IoT Hub</a>
+     */
+    @SuppressWarnings("unused")
```

Comment:
This method is for reference purposes, because people who are interested in this constructor will also be interested in learning how to create a shared access signature

Reply:
Can you add that as a comment in code?

Reply:
I did add that comment in the javadocs for this method